soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala

```diff
@@ -145,6 +145,7 @@ object SoQLFunctions {
 
   val BinaryPlus = f("+", SpecialFunctions.Operator("+"), Map("a" -> NumLike), Seq(VariableType("a"), VariableType("a")), Seq.empty, VariableType("a"))
   val BinaryMinus = f("-", SpecialFunctions.Operator("-"), Map("a" -> NumLike), Seq(VariableType("a"), VariableType("a")), Seq.empty, VariableType("a"))
+  val FloatingTimestampMinus = mf("floating timestamp -", SpecialFunctions.Operator("-"), Seq(SoQLFloatingTimestamp, SoQLFloatingTimestamp), Seq.empty, SoQLNumber)
```

Comment:
This will give nonsense answers a significant fraction of the time.

Reply:
(because it doesn't and cannot take into account DST shifts.  Put this operator on fixed timestamp and it will be fine.)

Reply:
We have very few fixed timestamp columns.  Doesn't it still make sense w/o time zone?  It can be wrong if people assume floating timestamp is fixed timestamp in a specific time zone.  But even that the error is contained and understood?  I think we may want to add two more functions though 1. the same function in fixed timestamp.  2. a function that takes a timezone and convert a floating timestamp to a fixed timestamp.

Reply:
I don't see better way to help people do this on columns that are already in floating_timestamp (the majority).

Reply:
It does not make sense without a time zone.  How many seconds are between noon on march 7th and noon on march 9th, in 2020?  The answer is: it depends on which time zone you're representing those times in.  If they're with respect to Seattle time, it's 47 hours, 169200 seconds.  If they're with respect to Tucson time, it's 48 hours, 172800 seconds.

Unfortunately, it's not actually possible to do this calculation in general given only a floating timestamp and a time zone, because the conversion between an instant and a wall-clock representation is lossy.  How many seconds are there between midnight at the start of the 1st of november 2020 and 1:30am on that same day, in Seattle specifically (so with the time zone "US/Pacific")?  The answer is: either 5400 seconds or 9000 seconds depending on whether you mean the first 1:30am on that day or the second 1:30am on that day, since it happens twice, and `floating_timestamp` does not represent that.

Really almost all times on the platform ought to be `fixed_timestamp`, with `floating_timestamp` reserved for presentation purposes ("I as a dataset owner want the times in this dataset to be displayed in New York time, not the user's local timezone, so I shall create a view with either `select ... to_floating_timestamp(my_time_column, 'US/Eastern') ...`" or a formatting option on the column that makes the frontend do that transformation either internally to itself or as a transformation on the soql it emits when populating its internal data.

Reply:
Perhaps if we only add math functions to fixed_timestamp and improve column formatting information so that it can include timezone or whatever we could entice people to use fixed_timestamp. Or make the migration ourselves.